main.rb

```diff
@@ -4,42 +4,48 @@
 require './book'
 require './rental'
 
-class Main
-  def initialize
-    @people = []
-    @books = []
-    @rentals = []
-  end
+class List
+  attr_reader :list
 
-  def print_options
-    puts 'Please choose an option by entering a number:'
-    puts '1- List all books'
-    puts '2- List all people'
-    puts '3- Create a person'
-    puts '4- Create a book'
-    puts '5- Create a rental'
-    puts '6- List all rentals for a given person id '
-    puts '7- exit'
+  def initialize(list, type)
+    @list = list
+    @type = type
   end
 
-  def list_books_or_people(input)
-    case input
-    when '1'
-      list_books
-    when '2'
-      list_people
+  def list_items
+    if @type == 'people'
+      @list.each_with_index { |item, i| puts "#{i + 1}: Name: #{item.name}, Age: #{item.age} ID: #{item.id}" }
+    else
+      @list.each_with_index { |item, i| puts "#{i + 1}: Title: #{item.title}, Author: #{item.author}" }
     end
   end
+end
 
-  def list_books
-    @books.each_with_index { |book, i| puts "#{i + 1}: Title: #{book.title}, Author: #{book.author}" }
-    puts
+class Books
```

Comment:
- [ ] Kindly extract this **Books** class into a different file in keeping with the SOLID rule.